packages/web3-eth-contract/src/utils.ts

```diff
@@ -227,6 +227,29 @@ export const getCreateAccessListParams = ({
 	return txParams;
 };
 
+/**
+ * Generates the Ethereum address of a contract created via a regular transaction.
+ *
+ * This function calculates the contract address based on the sender's address and nonce,
+ * following Ethereum's address generation rules.
+ *
+ * @param from The sender’s Ethereum {@link Address}, from which the contract will be deployed.
+ * @param nonce The transaction count (or {@link Numbers}) of the sender account at the time of contract creation.
+ *              This is used to ensure uniqueness of the generated address.
```

Comment:
I think this can be simplified as suggested below. Can you link to `Web3Eth.getTransactionCount` here since that is most likely how users will be acquiring the nonce? https://docs.web3js.org/api/web3/class/Web3Eth#getTransactionCount

```suggestion
 * @param nonce The transaction count (or {@link Numbers}) of the sender account at the time of contract creation.
```

Reply:
Of course, that would be clearer.